### DISCUSSION TITLE
How do I secure the docs route?

### DISCUSSION THREAD

**Discussion by NAN-CForadas** [apiflask/apiflask#365] [Q&A]
How do I secure the API Docs route?
```
openapi.docs                                                                                                                           GET      /docs
openapi.redoc                                                                                                                          GET      /redoc
```
I see how I can present my own 'specs' file in the documentation for [Register a spec processor] (https://apiflask.com/openapi/#register-a-spec-processor).  So I can put logic there for the spec to test for authenticated user.
```
openapi.spec                                                                                                                           GET      /openapi.json
```
My goal is to allow the teams developers to see the docs but not anyonte else.  Did I miss something in the docs?

**Comment by greyli** [comment]
Creating a custom view for docs should be suit your need: https://apiflask.com/api-docs/#add-custom-api-documentation

Maybe we can do something in APIFlask itself for this feature in the future.

**Reply by NAN-CForadas** [reply]
Thank you
